src/odb/src/lefin/lefTechLayerAreaRuleParser.cpp

```diff
@@ -0,0 +1,191 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <functional>
+#include <string>
+
+#include "boostParser.h"
+#include "db.h"
+#include "lefLayerPropParser.h"
+#include "lefin.h"
+
+namespace odb {
+
+lefTechLayerAreaRuleParser::lefTechLayerAreaRuleParser(lefin* l)
+{
+  lefin_ = l;
+}
+
+void lefTechLayerAreaRuleParser::parse(
+    std::string s,
+    odb::dbTechLayer* layer,
+    std::vector<std::pair<odb::dbObject*, std::string>>& incomplete_props)
+{
+  std::vector<std::string> rules;
+  boost::split(rules, s, boost::is_any_of(";"));
+  for (auto rule : rules) {
+    boost::algorithm::trim(rule);
+    if (rule.empty())
+      continue;
+    rule += " ; ";
+    if (!parseSubRule(rule, layer, incomplete_props))
+      lefin_->warning(278,
+                      "parse mismatch in layer propery LEF58_AREA for "
+                      "layer {} :\"{}\"",
+                      layer->getName(),
+                      rule);
+  }
+}
+
+void lefTechLayerAreaRuleParser::setInt(
+    double val,
+    odb::dbTechLayerAreaRule* rule,
+    void (odb::dbTechLayerAreaRule::*func)(int))
+{
+  (rule->*func)(lefin_->dbdist(val));
+}
+
+void lefTechLayerAreaRuleParser::setExceptEdgeLengths(
+    const boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptEdgeLengths(size);
+}
+
+void lefTechLayerAreaRuleParser::setExceptMinSize(
+    const boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptMinSize(size);
+}
+
+void lefTechLayerAreaRuleParser::setExceptStep(
+    const boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerAreaRule* rule)
+{
+  std::pair<int, int> size(lefin_->dbdist(at_c<0>(params)),
+                           lefin_->dbdist(at_c<1>(params)));
+  rule->setExceptStep(size);
+}
+
+void lefTechLayerAreaRuleParser::setTrimLayer(
+    std::string val,
+    odb::dbTechLayerAreaRule* rule,
+    odb::dbTechLayer* layer,
+    std::vector<std::pair<odb::dbObject*, std::string>>& incomplete_props)
+{
+  auto trim_layer = layer->getTech()->findLayer(val.c_str());
+  if (trim_layer != nullptr) {
+    rule->setTrimLayer(trim_layer);
+  } else {
+    incomplete_props.push_back({rule, val});
+  }
+}
+
+bool lefTechLayerAreaRuleParser::parseSubRule(
+    std::string s,
+    odb::dbTechLayer* layer,
+    std::vector<std::pair<odb::dbObject*, std::string>>& incomplete_props)
+{
+  qi::rule<std::string::iterator, std::string(), ascii::space_type> _string;
+  _string %= lexeme[+(char_ - ' ')];
+  odb::dbTechLayerAreaRule* rule = odb::dbTechLayerAreaRule::create(layer);
+
+  qi::rule<std::string::iterator, space_type> EXCEPTEDGELENGTH
```

Comment:
Use standard naming for variables (ie snake case).  Here and below

Reply:
@eder-matheus I think Matt meant to use snake case like: exceptEdgeLength. I personally don't mind the all caps because it matches the rule option name. We can fix this in a separate PR though because it's the same case for all parsers.